midterm-2023-11-06/code/5175-kenan-sipahi/script.ts

```diff
@@ -0,0 +1,378 @@
+document.addEventListener("DOMContentLoaded",siteCode);
+
+
+interface MetalBand{
+    id:number;
+    name:string;
+    genre:string;
+    formed:number;
+    location:string;
+    members:string[];
+    albums: Album[];
+
+}
+
+interface Album{
+    name:string;
+    year:number;
+}
+
+let bands : MetalBand[]=[];
+
+type BandSorter = (first: MetalBand,second: MetalBand) => number;
+
+
+
+
+const albumsSorter_asc : BandSorter = (first,second) =>second.albums.length- first.albums.length;
```

Comment:
Sorter methods for all the fields, and double at that :) +10
We can use some functional programming to actually make the descending sorts (they are just negatives of the ascending ones), i.e.

```javascript
const makeDescendingSorter = (ascSorter: BandSorter) => (first, second) => -ascSorter(first, second);

const albumsSorter_desc = makeDescendingSorter (albumsSorter_asc);
```
or maybe
```javascript
const makeDescendingSorter = (ascSorter: BandSorter) => (first, second) => ascSorter(second, first);
```

Reply:
Yeah i realized that while coding,this is very helpful.